### ISSUE TITLE
False Negative ``no-value-for-parameter`` for instances used inside of a class

### ISSUE THREAD

**Issue by bhmarscheck** [PyCQA/pylint#10731]
### Bug description

```python
# pylint: disable=too-few-public-methods
class Bird:
    def flap(self, wing):
        return wing

seagull = Bird()
print(seagull.flap())  # [no-value-for-parameter]

class BirdTester:
    def __init__(self, bird: Bird):
        self.bird = bird

    def bar1(self):
        bird: Bird = self.bird
        bird.flap()  # [no-value-for-parameter] false negative

    def bar2(self):
        self.bird.flap()  # [no-value-for-parameter] false negative

    def bar3(self, bird: Bird):
        bird.flap()  # [no-value-for-parameter] false negative
```

### Command used

```shell
pylint test.py
```

### Pylint output

```python
************* Module test
test.py:7:6: E1120: No value for argument 'a' in method call (no-value-for-parameter)
```

### Expected behavior

I would expect E1120 for all calles of `foo1` or at least for the calls inside bar1 and bar3

### Pylint version

```shell
pylint 4.0.2
astroid 4.0.2
Python 3.12.3 (tags/v3.12.3:f6650f9, Apr  9 2024, 14:05:25) [MSC v.1938 64 bit (AMD64)]
```

### OS / Environment

Windows 11

**Comment by Pierre-Sassoulas** [comment]
Indeed, thank you for opening the issue !
